### PR TITLE
Simplify and correct development documention

### PR DESCRIPTION
**Pull request by AntoinePrv**
- Use a single file for contributing guidelines
- Add doc builds to taskfile
- Fix sphinx warnings
- Simplify development doc

### CODE REVIEW

**Comment by AntoinePrv** [comment]
@dholth if that can be useful to you.

**Review by JohanMabille — APPROVED** [review]

**Comment by JohanMabille** [comment]
Documentation only, no need to wait for the CI.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+Contributing
+==============
 
 When contributing to this repository, it is always a good idea to first
 discuss the change you wish to make via issue, email, or any other method with
 the owners of this repository before making a change.
 
 We welcome all kinds of contribution -- code or non-code -- and value them
-highly. We pledge to treat everyone's contribution fairly and with respect and
+highly. We pledge to treat everyones contribution fairly and with respect and
```
Comment (dholth) [inline CONTRIBUTING.rst:10]:
```suggestion
highly. We pledge to treat everyone's contribution fairly and with respect and
```

The original is correct.

File: Taskfile.dist.yml [hunk 2]
```diff
@@ -48,24 +49,38 @@ tasks:
...
+    summary: |
+      Create development Conda environment with dependencies and development packages used to build
+      Mamba. Many tasks are automatically run inside this environment.
+      The environment is located at "{{.DEV_ENV_DIR}}" and can also be activated with micromamba to
+      benefit from the executables and LSP tools.
     cmds: [{task: '_create-env', vars: {prefix: '{{.DEV_ENV_DIR}}'}}]
 
   create-test-env:
-    summary: 'Copy the development environment'
+    desc: Create a local test environment with as a copy of the dev environment.
```
Comment (dholth) [inline Taskfile.dist.yml:61]:
```suggestion
    desc: Create a local test environment with a copy of the dev environment.
```

File: docs/source/developer_zone/build_locally.rst [hunk 3]
```diff
@@ -1,294 +0,0 @@
...
-
-.. code:: bash
-
-    cmake -B build/ \
-        -DBUILD_LIBMAMBA=ON \
-        -DBUILD_STATIC=ON \
-        -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX \
-        -DCMAKE_PREFIX_PATH=$CONDA_PREFIX \
-        -DBUILD_MICROMAMBA=ON
-    cmake --build build/
```
Comment (dholth) [inline docs/source/developer_zone/build_locally.rst:264]: these instructions were not that bad

**Pull request merged**